### DISCUSSION TITLE
gatsby build crashes on ./is-truthy utility

### DISCUSSION THREAD

**Discussion by karomancer** [gatsbyjs/gatsby#36530] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Upgraded Gatsby to the newest version (two major versions), and though I can run `gatsby develop` just fine, building crashes with the following error: 

```
$ gatsby build
internal/modules/cjs/loader.js:491
  throw new ERR_PACKAGE_PATH_NOT_EXPORTED(basePath, mappingKey);
  ^
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './is-truthy' is not defined by "exports" in /opt/build/repo/node_modules/gatsby-core-utils/package.json
    at applyExports (internal/modules/cjs/loader.js:491:9)
    at resolveExports (internal/modules/cjs/loader.js:507:23)
    at Function.Module._findPath (internal/modules/cjs/loader.js:635:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:953:27)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/opt/build/repo/node_modules/gatsby/node_modules/gatsby-cli/lib/reporter/reporter.js:34:17)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

I don't know if this has anything to do with it, but I found this chore commit moving the `is-truthy` utility that failed build checks:
https://github.com/gatsbyjs/gatsby/commit/837fcd2831cfaab38274466027b5384d4c19ad84

This is blocking me from full upgrading and deploying my site, so any workaround or advice would be much appreciated!

Thank you 🙇 

### Reproduction Link

https://github.com/karomancer/portfolio/pull/1

(sorry for the mess...definitely a WIP)

### Steps to Reproduce

I don't have great repro steps but
1. Upgrade 2 major versions of Gatsby
2. Dumb things down to accommodate upgrade (e.g. not using new image plugin yet)
3. Cannot build, but can develop
...

### Expected Result

Build successfully without errors about an internal util

### Actual Result

Error pasted above in description

### Environment

```shell
System:
    OS: macOS 12.4
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 17.7.2 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.8.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 104.0.5112.101
    Firefox: 104.0.1
    Safari: 15.5
  npmPackages:
    gatsby: ^4.13.1 => 4.22.0 
    gatsby-plugin-image: ^2.13.0 => 2.22.0 
    gatsby-plugin-react-helmet: ^5.13.0 => 5.13.0 
    gatsby-plugin-sass: ^5.13.0 => 5.13.0 
    gatsby-plugin-sharp: ^4.13.0 => 4.22.0 
    gatsby-plugin-sitemap: ^5.13.0 => 5.13.0 
    gatsby-source-contentful: ^7.11.0 => 7.20.0 
    gatsby-source-filesystem: ^4.13.0 => 4.22.0 
    gatsby-transformer-sharp: ^4.13.0 => 4.22.0 
  npmGlobalPackages:
    gatsby: 4.8.0
```

### Config Flags

No

**Comment by LekoArts** [comment]
Hi!

You must be running into an issue with version mismatches, sadly npm sometimes does that. I'd suggest to try the following:

1. Update the versions in the `package.json` all to their latest version (they are currently not)
2. Run `npm install --legacy-peer-deps`
3. Commit the package-lock.json file to the repo

**Reply by karomancer** [reply]
~Fixed it perfectly. Thank you so much! I wouldn't have been able to guess that.~

It fixed it on my Mac, but not on a linux box on Netlify. Still same error, all packages updated

**Comment by pixelsage** [comment]
Running into the same error with /.is-truthy! I'm trying to get my site deployed on Cloudflare, but this is preventing it from succeeding. Even after following the steps produced by @LekoArts above, I have the same error. (Can run/build fine on my M1 Mac, just not on Cloudflare).

**Reply by micduffy** [reply]
I am also encountering this issue on Netlify on Linux after upgrading the dependencies and with --legacy-peer-deps

**Comment by kieranstartup** [comment, marked as answer]
Hey,

I've just ran into this issue after upgrading from v2 to v4. I managed to fix it by upgrading my node version in Netlify to a more recent one. My site was launched 2 years ago and was running Node 10.x something old, and even though Netlify's default is 16.x, they don't automatically upgrade it after you launch a site. 

I tried creating a `.nvmrc` file with `16.16.0` as the value and it's finally building the site for me. So it might be worth trying this to see if it works!

**Reply by karomancer** [reply]
YOU SIR. Are my hero. 

Thank you so much. This actually fixed it!

**Reply by kieranstartup** [reply]
That's great @karomancer, glad to hear it helped!

I had to find a solution because this error totally broke a client's website, so I really had no option but to figure it out.
